src/Spice86.Core/Emulator/CPU/CfgCpu/ExecutionContextReturns.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace Spice86.Core.Emulator.CPU.CfgCpu;
+
+using Spice86.Shared.Emulator.Memory;
+using Spice86.Core.Emulator.CPU.CfgCpu.Linker;
+
+/// <summary>
+/// Maps return address to a stack of execution contexts to restore.
+/// Stack is because in some cases, code can call a new execution context that will return to the same address as the current one, so we need to know the call order to restore the correct one
+/// Cant be done with breakpoints since sometimes external int will happen even before the instruction after the iret is executed
```

Comment:
I don't understand this comment line ? Can it be expanded upon ?

Reply:
Tried my best to explain it, topic is no walk in a park.